workflows/dags/cds/cds_rdm_harvest.py

```diff
@@ -0,0 +1,134 @@
+    InspireHTTPRecordManagementHook,
+)
+from include.utils.alerts import task_failure_alert
+from include.utils.cds import (
+    get_identifiers_for_scheme,
+    has_any_rdm_id,
+    retrieve_and_validate_record,
+    update_record,
+)
+from inspire_utils.helpers import force_list
+from inspire_utils.record import get_value
+
+logger = logging.getLogger(__name__)
+
+
+@dag(
+    start_date=datetime.datetime(2025, 5, 22),
+    schedule="0 4 * * *",
+    catchup=False,
+    tags=["cds_rdm"],
+    params={
+        "since": Param(type=["null", "string"], default=""),
+    },
+    on_failure_callback=task_failure_alert,
+)
+def cds_rdm_harvest_dag():
+    """Defines the DAG for the CDS harvest workflow."""
+    generic_http_hook = GenericHttpHook(http_conn_id="cds_rdm_connection")
+    inspire_http_record_management_hook = InspireHTTPRecordManagementHook()
+
+    @task(task_id="get_cds_rdm_data")
+    def get_cds_rdm_data(**context):
+        """Starts harvesting cds rdm.
+
+        Returns: data from cds rdm
+        """
+        since = context["params"]["since"] or context["ds"]
+
+        all_hits = []
+
+        params = {
+            "q": f"updated:>={since}",
+            "page": 1,
+            "size": 50,
+            "sort": "newest",
+        }
+
+        while True:
+            logger.info(f"Harvesting CDS RDM data since {since}")
+            resp = generic_http_hook.call_api(
+                endpoint="/api/records", method="GET", params=params
+            )
+            resp.raise_for_status()
+            payload = resp.json()
+
+            hits = payload.get("hits", {}).get("hits", [])
+            logger.info(f"Fetched {len(hits)} records from CDS")
+
+            filtered = [h for h in hits if has_any_rdm_id(h)]
+            logger.info(f"{len(filtered)} records passed filter")
+            all_hits.extend(filtered)
+
+            next_url = payload.get("links", {}).get("next")
+            if not next_url:
+                logger.info("No more pages; finished harvesting.")
+                break
+
+            parsed = urlparse(next_url)
+            params = {k: v[0] for k, v in parse_qs(parsed.query).items()}
+
+            logger.info(f"Following next link: {next_url}")
+
+        logger.info(f"Total harvested records (post-filter): {len(all_hits)}")
+        return all_hits
+
+    @task_group
+    def process_cds_rdm_response(cds_record):
+        @task(task_id="process_record")
+        def process_record(cds_record):
+            cds_id = cds_record.get("id")
+            identifiers = get_value(cds_record, "metadata.identifiers", [])
+            control_numbers = get_identifiers_for_scheme(identifiers, "inspire")
+            arxivs = get_identifiers_for_scheme(identifiers, "arxiv")
+            dois = force_list(
+                get_value(cds_record, "pids.doi.identifier", [])
+            ) + force_list(get_value(cds_record, "parent.pids.doi.identifier", []))
```

Comment:
can we use the `has_any_rdm_id`?

Reply:
we also need these values here. The has_any_rdm_id is just used for prefiltering. I have extracted dois get in a separate function